drift-transport-netty/src/test/java/com/facebook/drift/transport/netty/codec/TestHeaderTransport.java

```diff
@@ -21,29 +21,51 @@
 import com.facebook.drift.protocol.TProtocolWriter;
 import com.facebook.drift.transport.netty.buffer.TestingPooledByteBufAllocator;
 import com.facebook.drift.transport.netty.ssl.TChannelBufferOutputTransport;
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufAllocator;
 import org.testng.annotations.Test;
 
+import java.util.List;
 import java.util.Optional;
 
 import static com.facebook.drift.protocol.TMessageType.CALL;
 import static com.facebook.drift.protocol.TMessageType.ONEWAY;
+import static com.facebook.drift.transport.netty.codec.HeaderTransport.Transform;
+import static com.facebook.drift.transport.netty.codec.HeaderTransport.Transform.ZLIB_TRANSFORM;
+import static com.facebook.drift.transport.netty.codec.HeaderTransport.decodeFrame;
+import static com.facebook.drift.transport.netty.codec.HeaderTransport.encodeFrame;
 import static com.facebook.drift.transport.netty.codec.HeaderTransport.tryDecodeFrameInfo;
 import static com.facebook.drift.transport.netty.codec.Protocol.BINARY;
 import static com.facebook.drift.transport.netty.codec.Protocol.FB_COMPACT;
 import static com.facebook.drift.transport.netty.codec.Transport.HEADER;
 import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
 
 public class TestHeaderTransport
 {
     @Test
-    public void testTryDecodeSequenceId()
+    public void testNoTransform()
+            throws Exception
+    {
+        assertTryDecodeSequenceId(ImmutableList.of());
+        assertRoundTrip(ImmutableList.of());
+    }
+
+    @Test
+    public void testZlibTransform()
+            throws Exception
+    {
+        assertTryDecodeSequenceId(ImmutableList.of(ZLIB_TRANSFORM));
+        assertRoundTrip(ImmutableList.of(ZLIB_TRANSFORM));
+    }
+
+    private void assertTryDecodeSequenceId(List<Transform> transforms)
```

Comment:
Add a test case when message is partially decoded. e.g.: https://gist.github.com/arhimondr/73e485aedb66b5fcf53a86335b4278da